api/hub/src/datasources/moderation-report.ts

```diff
@@ -0,0 +1,162 @@
+import { DataSource } from 'apollo-datasource';
+import { getAppDB, logger, decodeString, encodeString } from '../helpers';
+import { Client, ThreadID, Where } from '@textile/hub';
+import { ModerationDecision, ModerationReport } from '../collections/interfaces';
+import ModerationDecisionAPI from './moderation-decision';
+import { queryCache } from '../storage/cache';
+
+class ModerationReportAPI extends DataSource {
+  private readonly collection: string;
+  private context: any;
+  private readonly dbID: ThreadID;
+  constructor({ collection, dbID }) {
+    super();
+    this.collection = collection;
+    this.dbID = dbID;
+  }
+
+  async initialize(config) {
+    this.context = config.context;
+  }
+
+  async countReports(contentID: string) {
+    const db: Client = await getAppDB();
+    const query = new Where('contentID').eq(contentID);
+    const reports = await db.find<ModerationReport>(this.dbID, this.collection, query);
+    return reports.length;
+  }
+
+  async exists(contentID: string, author: string) {
+    const db: Client = await getAppDB();
+    const query = new Where('contentID').eq(contentID).and('author').eq(author);
+    const report = await db.find<ModerationReport>(this.dbID, this.collection, query);
+    if (report.length) {
+      return true;
+    }
+    return false;
+  }
+
+  async listReports(contentID: string, offset?: number, limit?: number) {
+    const db: Client = await getAppDB();
+    const query = new Where('contentID')
+      .eq(contentID)
+      .skipNum(offset || 0)
+      .limitTo(limit || 10)
+      .orderByDesc('creationDate');
+    const reports = await db.find<ModerationReport>(this.dbID, this.collection, query);
+    const list = [];
+    for (const report of reports) {
+      report.explanation = decodeString(report.explanation);
+      list.push(report);
+    }
+    return list;
+  }
+
+  async getReport(contentID: string, author: string) {
```

Comment:
some comments on each method in the ts-doc format would be nice :D

Reply:
You are right. I'll add these before merging.